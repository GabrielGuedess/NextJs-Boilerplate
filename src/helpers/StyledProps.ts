export type StyledProps<T> = Required<{
  [K in keyof T as `$${string & K}`]: T[K];
}>;
