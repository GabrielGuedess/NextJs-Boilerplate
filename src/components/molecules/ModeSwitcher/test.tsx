import { ThemeProvider } from 'next-themes';

import userEvent from '@testing-library/user-event';
import { screen, waitFor } from '@testing-library/react';

import { render } from 'utils/tests/helpers';

import { ModeSwitcher } from '.';

describe('<ModeSwitcher />', () => {
  it('should render the mode switcher correctly', async () => {
    const { container } = render(
      <ThemeProvider>
        <ModeSwitcher />
      </ThemeProvider>,
      { theme: 'system' },
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the mode switcher light', async () => {
    render(
      <ThemeProvider>
        <ModeSwitcher />
      </ThemeProvider>,
      { theme: 'light' },
    );

    const button = screen.getByLabelText('Mode Switcher');

    expect(button.getAttribute('aria-checked')).toBe('false');

    await userEvent.click(button);

    await waitFor(() => {
      expect(button.getAttribute('aria-checked')).toBe('true');
    });
  });

  it('should render the mode switcher dark', async () => {
    render(
      <ThemeProvider>
        <ModeSwitcher />
      </ThemeProvider>,
      { theme: 'dark' },
    );

    const button = screen.getByLabelText('Mode Switcher');

    expect(button.getAttribute('aria-checked')).toBe('true');

    await userEvent.click(button);

    await waitFor(() => {
      expect(button.getAttribute('aria-checked')).toBe('false');
    });
  });
});
