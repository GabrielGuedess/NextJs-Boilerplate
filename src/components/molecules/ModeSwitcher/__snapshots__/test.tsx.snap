// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModeSwitcher /> should render the mode switcher correctly 1`] = `
<script
  nonce=""
>
  ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system","light",["light","dark"],null,false,true)
</script>
`;
